pkg/kapp/diff/change_precalculated.go

```diff
@@ -51,9 +51,10 @@ func (d *ChangePrecalculated) NewOrExistingResource() ctlres.Resource {
 	panic("Not possible")
 }
 
-func (d *ChangePrecalculated) NewResource() ctlres.Resource      { return d.newRes }
-func (d *ChangePrecalculated) ExistingResource() ctlres.Resource { return d.existingRes }
-func (d *ChangePrecalculated) AppliedResource() ctlres.Resource  { return d.appliedRes }
+func (d *ChangePrecalculated) NewResource() ctlres.Resource             { return d.newRes }
+func (d *ChangePrecalculated) ExistingResource() ctlres.Resource        { return d.existingRes }
+func (d *ChangePrecalculated) AppliedResource() ctlres.Resource         { return d.appliedRes }
+func (d *ChangePrecalculated) ClusterOriginalResource() ctlres.Resource { return d.existingRes }
```

Comment:
Will this ever be called? Is the existing resource for a recalculated change not rebased?

If we never run into this code path for recalculated change, should call that out here instead of returning the existing res?

Reply:
> should call that out here instead of returning the existing res?

Do you mean we should just add a comment?

Reply:
I was thinking more along the lines of `panic("Unimplemented")` and a comment. So that no one down the line does not use it expecting a cluster original resource. We won't hit that if the current code does not. Or maybe `return nil` at the least because we do not actually have that information.

Reply:
Sure, that makes sense.

Reply:
This will be required for all the changes, it will be used in change summary.